File: app/models/production_order.py
from sqlalchemy import Column, Integer, String, TIMESTAMP, Enum as SQLAlchemyEnum, DECIMAL, ForeignKey
from sqlalchemy.orm import relationship
from app.database.base import Base

class ProductionOrder(Base):
    __tablename__ = "production_orders"
    id = Column(Integer, primary_key=True, index=True)
    lot_number = Column(String(100), unique=True, nullable=False)
    part_number = Column(String(100), ForeignKey("master_prod.part_number"))
    quantity_to_produce = Column(DECIMAL(10, 2), nullable=False)
    initial_wip_stock = Column(DECIMAL(10, 2), nullable=False)
    status = Column(SQLAlchemyEnum('running', 'rework', 'pending', 'cancelled', name='po_status_enum'), nullable=False)
    created_by_id = Column(Integer, ForeignKey("users.id"))
    start_date = Column(TIMESTAMP)
    completion_date = Column(TIMESTAMP)
    created_at = Column(TIMESTAMP)
    
    creator = relationship("User")
    product = relationship("MasterProd")

File: node/models/productionOrder.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const ProductionOrder = sequelize.define('ProductionOrder', {
    id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
    lot_number: { type: DataTypes.STRING(100), allowNull: false, unique: true },
    part_number: { type: DataTypes.STRING(100), allowNull: false },
    quantity_to_produce: { type: DataTypes.DECIMAL(10, 2), allowNull: false },
    initial_wip_stock: { type: DataTypes.DECIMAL(10, 2), allowNull: false },
    status: { type: DataTypes.ENUM('running', 'rework', 'pending', 'cancelled'), defaultValue: 'running' },
    created_by_id: { type: DataTypes.INTEGER },
    start_date: { type: DataTypes.DATE },
    completion_date: { type: DataTypes.DATE },
    created_at: { type: DataTypes.DATE, defaultValue: DataTypes.NOW }
}, { tableName: 'production_orders', timestamps: true });

module.exports = ProductionOrder;
